docs/data types.md

```diff
@@ -0,0 +1,277 @@
+
+Example:
+
+```jsx
+const sym = Symbol('foo');
+```
+
+## Composite Data Types
+
+### Arrays
+
+Think of arrays as a collection of your favorite books. 1. You have a list of titles: "Harry Potter," "Percy Jackson," and "The Hunger Games." In JavaScript, you can use an array to store these book titles together.
+
+Items in arrays are called ***“Elements”***
+
+JavaScript arrays are written with square brackets. Elements are separated by commas.
+
+```jsx
+const my_book_collection = ["Harry Potter", "Percy Jackson", "The Hunger Games"];
+
+const cars = ["Saab", "Volvo", "BMW"];
+```
+
+*Note: the element can be of any type, it does not necessarily have to be string as shown in the example.*
+
+Array indexes are zero-based, which means the first item is [0], second is [1], and so on.
+
+In the above example:
+
+```jsx
+cars[0] // Saab
+cars[1] // Volvo
+cars[2] // BMW
+```
+
+Further explanations are available in the ***Arrays*** chapter!
+
+### Objects
+
+Consider a recipe for a sandwich. It has different parts: "ingredients," "steps," and "prep time." In JavaScript, you can create an object to hold all these details in a structured way.
+
+JavaScript objects are written with curly braces `{}`
+
+Object properties are written as name: value pairs, separated by commas.
+
+Example:
+
+```jsx
+const sandwich = {
+		name: "Ham Cheese Sandwich",
+		ingredients: ["bread", "ham", "cheese"],
+		steps: [......],
+		preptime: ....
+}
+```
+
+An easier example would be an object of a person
+
+```jsx
+const person = {
+		firstName: "John",
+		lastName: "Doe",
+		age: 50,
+		eyeColor: "blue"
+}
+```
+
+The object (person) in the example above has 4 properties: firstName, lastName, age, and eyeColor.
+
+With the defined properties, we can access them easily via
+
+```jsx
+console.log(person.firstName) // "John"
+console.log(person.lastName) // "Doe"
+```
+
+As well as manipulating them
+
+```jsx
+console.log(person.firstName) // Currently is "John"
+
+person.firstName = "Christopher"
+
+console.log(person.firstname) // Will now become "Christopher"
+```
+
+What if you try to manipulate a non-existing property?
+
+```jsx
+person.phoneNumber = 1235666;
+```
+
+The new property `phoneNumber` will be added to that object automatically!
+
+Try it yourself?
+
+- Create an object representing a book with properties like **`title`**, **`author`**, and **`year`**.
+- Add a new property **`genre`** to the object and set its value.
+- Access and print the author's name from the object.
+
+<details>
+<summary>Answer</summary>
+<p>
+
+```jsx
+    const book = {
+    		title: "Harry Potter",
+    		author: "J K Rolling",
+    		year: 1997
+    };
+    
+    book.genre = "fantasy"
+    
+    console.log(book.author)
+```
+</p>    
+</details>
+
+## The typeof Operator
+
+You can use the JavaScript typeof operator to find the type of a JavaScript variable.
+
+The typeof operator returns the type of a variable or an expression:
```

Comment:
it returns a string containing the type, not an actual type

Reply:
okie